src/LegacySupport/SystemIndex/Index.cs

```diff
@@ -0,0 +1,160 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics.CodeAnalysis;
+using System.Runtime.CompilerServices;
+
+#pragma warning disable CS0436 // Type conflicts with imported type
+#pragma warning disable S3427 // Method overloads with default parameter values should not overlap 
+#pragma warning disable SA1642 // Constructor summary documentation should begin with standard text
+#pragma warning disable IDE0011 // Add braces
+#pragma warning disable SA1623 // Property summary documentation should match accessors
+#pragma warning disable IDE0023 // Use block body for conversion operator
+#pragma warning disable S3928 // Parameter names used into ArgumentException constructors should match an existing one
+#pragma warning disable LA0001 // Use the 'Microsoft.Shared.Diagnostics.Throws' class instead of explicitly throwing exception for improved performance
+#pragma warning disable CA1305 // Specify IFormatProvider
+
+namespace System
+{
+    internal readonly struct Index : IEquatable<Index>
```

Comment:
This is fine for now. But we also shipped Index and Range in a downlevel package, didn't we?

Reply:
TIL, seems we ship it with Microsoft.Bcl.Memory. It's only available as a 9.0.x package which create complaints from folks avoiding odd major version numbers.